core/src/com/ericc/the/game/utils/GridPoint.java

```diff
@@ -0,0 +1,45 @@
+package com.ericc.the.game.utils;
+
+public class GridPoint {
+    public final int x;
+    public final int y;
+
+    public GridPoint(int x, int y) {
+        this.x = x;
+        this.y = y;
+    }
+
+    public GridPoint add(GridPoint other) {
+        return new GridPoint(this.x + other.x, this.y + other.y);
+    }
+
+    public GridPoint subtract(GridPoint other) {
+        return new GridPoint(this.x - other.x, this.y - other.y);
+    }
+
+    public GridPoint shift(int x, int y) {
+        return new GridPoint(this.x + x, this.y + y);
```

Comment:
shift moze skorzystac z add, redukujemy wtedy możliwe błedy, bo przeciez ta funkcja to tylko ladny alias